dimod/binary/binary_quadratic_model.py

```diff
@@ -758,6 +758,93 @@ def add_linear_equality_constraint(
                     u, v, 2 * lagrange_multiplier * ubias * vbias)
         self.offset += lagrange_multiplier * constant * constant
 
+    def add_linear_inequality_constraint(
+            self, terms: Iterable[Tuple[Variable, Bias]],
+            lagrange_multiplier: Bias,
+            label: str,
+            constant: int = 0,
+            lb: int = - np.Infinity,
+            ub: int = 0,
+            slack_method: str = "log2",
+            cross_zero: bool = False
+    ):
+        """Add a linear inequality constraint as a quadratic objective.
+
+        Adds a linear inequality constraint of the form:
+
+        math:'lb <= \sum_{i,k} a_{i,k} x_{i,k} + constant <= ub'
+        to the binary quadratic model as a quadratic objective.
+        For constraints with fractional coefficients, multiply both sides of the inequality by an
```

Comment:
If this is a requirement, then we need to enforce it in code and raise an exception or at the very least a warning. Users don't read docstrings :smile:

Reply:
I'll add a warning. The formulation/method when there are fractional values are approximate and not necessarily invalid.